### DISCUSSION TITLE
Output JSON objects from YOLOv8 CLI tool

### DISCUSSION THREAD

**Discussion by trevorstr** [ultralytics/ultralytics#9518] [General]
Hello, I am using the official container image for YOLOv8 directly from the Docker Hub, without any special build process for a custom image. I'm using it to perform object detection against a variety of RTSP streams, directly streaming video footage from security cameras. I have the NVIDIA Container Toolkit installed on my bare metal server, which is equipped with an NVIDIA GeForce RTX 3060 12 GB GPU.

I simply launch a new container, using the `ultralytics/ultralytics` container image, similar to the following:

```
docker run --restart=always ultralytics/ultralytics bash -c 'curl -LO <yolov8l_url>; yolo predict rtsp://<url> conf=0.8 imgsz=1920 model=yolov8l.pt'
```

The Docker Engine handles storing and rotating logs locally by default. I wrote an external process, using PowerShell script, that queries the Docker logs, for each Ultralytics container, and processes several lines of data, looking for certain COCO object classes in the output. I have to use a regex right now, since the output is not in a standard format.

It would be much simpler to write integrations around the YOLOv8 CLI tool if it were able to produce JSON output.

Is there a parameter I can specify, on the CLI, that will tell it to output JSON Lines, instead of the default output format?

**Comment by pderrenger** [comment]
Hello! 🚀 Thanks for reaching out with your inquiry about YOLOv8 CLI tool output format.

Currently, the YOLOv8 CLI does not directly support converting its standard output format to JSON Lines through a standalone CLI parameter. However, you can achieve this functionality by redirecting the output to a Python script for parsing and formatting.

Here's a compact example that demonstrates how you could use Python for converting output to JSON Lines:

```bash
# Assuming yolov8_output_parser.py is your Python script to parse and convert output
docker run --restart=always ultralytics/ultralytics bash -c 'curl -LO <yolov8l_url>; yolo predict rtsp://<url> conf=0.8 imgsz=1920 model=yolov8l.pt' | python yolov8_output_parser.py
```

And your `yolov8_output_parser.py` might look something like this (pseudo):

```python
import sys
import json

# This is a very basic example. You might need to adjust it based on actual output format.
for line in sys.stdin:
    # Parse the line using regex or any preferred method
    parsed_output = parse_line(line) # Implement this function based on your needs
    # Convert the parsed output to JSON
    json_output = json.dumps(parsed_output)
    print(json_output)
```

We value your feedback and continuously work on enhancing YOLOv8 to meet our community's needs. It's a great suggestion to have native JSON line output support for easier integration in future releases! 🌟

Let us know if you have any additional questions!

**Comment by trevorstr** [comment]
Thanks, Paula. Since I am already parsing the data with PowerShell code, I will just handle it there. Right now, I am just using a simple regular expression to pull our data, ie. `\d car|\d truck` to detect the number of cars or trucks. It's not ideal, but it works fine.

I was just hoping that the CLI itself could spit out JSON blobs instead of the basic output it currently has, for ease of integration.

I appreciate the answer! 💯

**Reply by glenn-jocher** [reply]
Hey there! I understand your situation, and it's quite inventive to use PowerShell and regex for parsing the outputs 🧙‍♂️. I'm glad you've found a workaround that suits your needs. While the CLI doesn't currently support direct JSON output, your feedback is valuable and something we might consider for future updates. Thanks for being a part of the community and sharing your approach! 🚀 Keep rocking with your integrations!
